src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.IntTC.cs

```diff
@@ -11,9 +11,21 @@ namespace Microsoft.CodeAnalysis.CSharp
 
     internal static partial class ValueSetFactory
     {
-        private struct IntTC : INumericTC<int>
+        private class IntTC : INumericTC<int>
         {
-            int INumericTC<int>.MinValue => int.MinValue;
+            // Note: whenever we intersect or union two sets of IntTCs,
+            // we just keep the nonNegative flag of the set we're merging into.
+            public bool nonNegative;
```

Comment:
>public bool nonNegative; [](http://example.com/codeflow?start=12&length=24)

I really like the fact that we no longer have to use different types to represent this difference in behavior